libvirt/tests/src/cpu/vcpupin.py

```
@@ -9,13 +9,38 @@
 from virttest import utils_hotplug
 
 
+def check_vcpu_placement(test, params):
+    """
+    return 1 if the value of placement or cpuset is not expected
+    """
+
+    vm_name = params.get("main_vm")
+    numatune_memory_mode = params.get("numatune_memory_mode", "")
+    numatune_memory_placement = params.get("numatune_memory_placement", "")
+
+    vmxml = vm_xml.VMXML.new_from_dumpxml(vm_name)
+    logging.debug(vmxml)
+
+    if vmxml.placement == "static":
+        if vmxml.cpuset == "1":
+            return 0
```

suggest directly do test.fail  here ,rather than check the return value

---

updated